export-script/generate_ror_id_matches.py

```diff
@@ -0,0 +1,126 @@
+import os
+import psycopg2
+import requests
+import csv
+import logging
+import time
+from concurrent.futures import ThreadPoolExecutor, as_completed
+from dotenv import load_dotenv
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+
+# Load environment variables
+project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+dotenv_path = os.path.join(project_root, '.env')
+load_dotenv(dotenv_path, override=True)
+
+# Database connection parameters
+conn_params = {
+    'dbname': os.getenv('DB_NAME'),
+    'user': os.getenv('DB_USER'),
+    'password': os.getenv('PG_PASSWORD'),
+    'host': os.getenv('DB_HOST'),
+    'port': os.getenv('DB_PORT')
+}
+
+# Connect to the database
+conn = psycopg2.connect(**conn_params)
+cursor = conn.cursor()
+
+# Query to get organization names without external_id
+query = """
+SELECT id, title
+FROM affiliations
+WHERE external_id IS NULL
+UNION
+SELECT id, title
+FROM funders
+WHERE external_id IS NULL;
+"""
+
+logging.info("Executing database query to retrieve organizations...")
+cursor.execute(query)
+organizations = cursor.fetchall()
+total_records = len(organizations)
+logging.info(f"Total organizations to process: {total_records}")
+
+# ROR API endpoint
+ROR_API_URL = "https://api.ror.org/organizations?affiliation="
+
+# Function to query ROR API
+def query_ror_api(organization_name):
+    try:
+        response = requests.get(ROR_API_URL + requests.utils.quote(organization_name))
```

Comment:
ElasticSearch reserved chars should be escaped bc some will cause 500 errors (like unclosed quotes or parens) and others will just have unexpected effects on the query results. Escape reserved chars by placing using URL encoded backslash  %5C before each (URL encoded) reserved char. https://ror.readme.io/docs/api-query#special-characters

Reply:
Done